### ISSUE TITLE
False positive CA2208 in extension block

### ISSUE THREAD

**Issue by andryushchenko** [dotnet/roslyn-analyzers#7774]
### Describe the bug

False positive [CA2208](https://learn.microsoft.com/dotnet/fundamentals/code-analysis/quality-rules/ca2208) for extension block argument if extension method has other arguments

### To Reproduce

Simple code to reproduce with .NET SDK 10.0.100

```
public static class ExtensionTest
{
    extension(object item)
    {
        public void Test1()
        {
            _ = item ?? throw new ArgumentNullException(nameof(item)); // OK
        }
        
        public void Test2(object argument)
        {
            _ = item ?? throw new ArgumentNullException(nameof(item)); // CA2208
            _ = argument ?? throw new ArgumentNullException(nameof(argument)); // OK
        }
    }
}
```

### Expected behavior
No warnings
